web/html/src/components/formulas/FormulaComponentGenerator.js

```diff
@@ -1,69 +1,87 @@
 /* eslint-disable */
-// needed because of circular deps
-module.exports = {
-  generateFormulaComponent: generateFormulaComponent,
-  generateFormulaComponentForId: generateFormulaComponentForId
-}
+import * as React from 'react';
+import EditGroup from './EditGroup';
+import Group from './Group';
+import PasswordInput from './PasswordInput';
+import {default as Jexl} from 'jexl';
+import HelpIcon from 'components/HelpIcon';
 
-const React = require("react");
-const EditGroup = require("./EditGroup").EditGroup;
-const PasswordInput = require("./PasswordInput").PasswordInput;
 const BASIC_INPUT_TYPES = ["text", "email", "url", "date", "time"];
 
 
-function generateFormulaComponent(element, value, formulaForm, parents, wrapper) {
+export function generateFormulaComponent(element, value, formulaForm, parents, wrapper, disabled = false) {
     const id = (parents ? parents + "#" : "") + element.$id;
-    return generateFormulaComponentForId(element, value, formulaForm, id, wrapper);
+    return generateFormulaComponentForId(element, value, formulaForm, id, wrapper, disabled);
 }
 
-function generateFormulaComponentForId(element, value, formulaForm, id, wrapper) {
+export function generateFormulaComponentForId(element, value, formulaForm, id, wrapper, disabled = false) {
     wrapper = get(wrapper, defaultWrapper);
 
-    var isDisabled = (formulaForm.props.scope !== element.$scope && element.$scope !== "system");
+    var isDisabled = (formulaForm.props.scope !== element.$scope && element.$scope !== "system") || 
+        ("$disabled" in element && evalExpression(id, element.$disabled, formulaForm)) ||
+        (!("$disabled" in element) && disabled);
+
+    if ("$visible" in element && !evalExpression(id, element.$visible, formulaForm)) {
+        return null;
+    }
 
+    // $visibleIf is deprecated
```

Comment:
This would be especially important to mention it in the developers documentation. By the way, we should also link the development docs from the `salt-formulas` repo README section.

Reply:
I'll update the developer documentation after this PR is merged.